obelisk/cpp/zoo/hardware/sensing/zed/zed2_sensors_interface.h

```diff
@@ -0,0 +1,420 @@
+#pragma once
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <fstream>
+#include <msg_conversions.h>
+#include <mutex>
+#include <obelisk_sensor_msgs/msg/obk_image.hpp>
+#include <obelisk_std_msgs/msg/u_int8_multi_array.hpp>
+#include <rclcpp/callback_group.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/timer.hpp>
+#include <rclcpp_lifecycle/lifecycle_node.hpp>
+#include <set>
+#include <sl/Camera.hpp>
+#include <stdexcept>
+#include <unsupported/Eigen/CXX11/Tensor>
+#include <yaml-cpp/yaml.h>
+
+#include "obelisk_sensor.h"
+
+/**
+ * @brief Class for managing ZED2 cameras using the Obelisk framework.
+ */
+class ObeliskZed2Sensors : public obelisk::ObeliskSensor {
+
+  public:
+    /**
+     * @brief Construct a new ObeliskZed2Sensors object.
+     *
+     * @param node_name Name of the ROS2 node.
+     */
+    ObeliskZed2Sensors(const std::string& node_name) : obelisk::ObeliskSensor(node_name) {
+        // Declare parameters
+        this->declare_parameter("params_path_pkg", "");
+        this->declare_parameter("pub_period", 0.0);
+
+        // Get parameter values
+        pub_period_                 = this->get_parameter("pub_period").as_double();
+        std::string params_path_pkg = this->get_parameter("params_path_pkg").as_string();
+        std::string _params_path    = this->get_parameter("params_path").as_string();
+
+        std::string params_path;
+        if (_params_path[0] == '/') {
+            params_path = _params_path;
+        } else if (params_path_pkg.empty()) {
+            std::string err_msg = "No package provided for ZED2 camera parameters!";
+            RCLCPP_ERROR(this->get_logger(), err_msg.c_str());
```

Comment:
There is technically nothing wrong with this, but I think its cleaner to just use `RCLCPP_ERROR_STREAM` then you can just easily put the message in as a normal stream. I.e. you don't have to make a string in the line above. Tbh even with the non-stream option I doubt you need that string, and it can prob be just written in the error line.

Reply:
The main reason I do this is because I not only want to display this error, I also want to pass the error message into a runtime error that I throw. Therefore, it's cleaner for me to define the error message string and then pass to both.